libr/core/cmd_anal.c

```diff
@@ -6201,20 +6218,31 @@ static void __anal_esil_function(RCore *core, ut64 addr) {
 			while (pc < end) {
 				left = R_MIN (end - pc, 32);
 				// r_asm_set_pc (core->rasm, pc);
-				ret = r_anal_op (core->anal, &op, pc, buf + pc - bb->addr, left, R_ANAL_OP_MASK_HINT | R_ANAL_OP_MASK_ESIL); // read overflow
+				ret = r_anal_op (core->anal, &op, pc, buf + pc - bb->addr, left, R_ANAL_OP_MASK_HINT | R_ANAL_OP_MASK_ESIL| R_ANAL_OP_MASK_DISASM); // read overflow
 				opskip = false;
 				switch (op.type) {
+				case R_ANAL_OP_TYPE_JMP:
+				case R_ANAL_OP_TYPE_NOP:
+				case R_ANAL_OP_TYPE_CJMP:
+				case R_ANAL_OP_TYPE_UJMP:
 				case R_ANAL_OP_TYPE_CALL:
 				case R_ANAL_OP_TYPE_RET:
 					opskip = true;
 					break;
 				}
 				if (ret) {
-					if (opskip) {
-						r_reg_set_value_by_role (core->anal->reg, R_REG_NAME_PC, pc);
-						r_anal_esil_parse (core->anal->esil, R_STRBUF_SAFEGET (&op.esil));
-						r_anal_esil_dumpstack (core->anal->esil);
-						r_anal_esil_stack_free (core->anal->esil);
+					if (!opskip) {
+						const char *esilstr = R_STRBUF_SAFEGET (&op.esil);
+						// eprintf ("0x%08"PFMT64x"  %s\n", pc, op.mnemonic);
+						if (R_STR_ISNOTEMPTY (esilstr)) {
+							r_anal_esil_set_pc (core->anal->esil, pc);
+							r_reg_set_value (core->anal->reg, r_reg_get (core->anal->reg, "pc", -1), pc);
```

Comment:
is the ```r_reg_get ``` correct here, or does it only work, if the pc-register is named "pc"?

Reply:
yeah that was just a PoC for the specific use case. ill push tests for this now